mindmap/mindmap.py

```diff
@@ -93,20 +108,81 @@ def studio_view(self, context=None):
         frag.initialize_js("MindMapXBlock")
         return frag
 
-    # TO-DO: change this handler to perform your own actions.  You may need more
-    # than one handler, or you may not need any handlers at all.
-    @XBlock.json_handler
-    def increment_count(self, data, suffix=''):
+    def connect_to_s3(self):
+        """
+        Create a connection to S3.
+
+        Returns:
+            boto3.client: The connection to S3.
         """
-        An example handler, which increments the data.
+        AWS_ACCESS_KEY_ID = getattr(settings, "AWS_ACCESS_KEY_ID", None)
+        AWS_SECRET_ACCESS_KEY_ID = getattr(settings, "AWS_SECRET_ACCESS_KEY_ID", None)
+
+        return boto3.client(
+            "s3",
+            aws_access_key_id=AWS_ACCESS_KEY_ID,
+            aws_secret_access_key=AWS_SECRET_ACCESS_KEY_ID,
+        )
+
+    def get_key(self) -> str:
+        """
+        Return the key (path) to use in S3.
+
+        Returns:
+            str: The key (path) to use in S3.
         """
-        if suffix:
-            pass  # TO-DO: Use the suffix when storing data.
-        # Just to show data coming in...
-        assert data['hello'] == 'world'
+        user_service = self.runtime.service(self, "user")
+        user = user_service.get_current_user()
+        anonymous_user_id = self.anonymous_user_id(user)
+        block_id = self.location.block_id # pylint: disable=no-member
+        return f"{block_id}/{anonymous_user_id}/mindmap.json"
 
-        self.count += 1
-        return {"count": self.count}
+    def file_exists(self) -> bool:
+        """
+        Check if the file exists in S3.
+
+        Returns:
+            bool: True if the file exists, False otherwise.
+        """
+        s3_client = self.connect_to_s3()
+        try:
+            s3_client.head_object(Bucket=AWS_BUCKET_NAME, Key=self.get_key())
+        except s3_client.exceptions.ClientError as e:
+            if e.response["Error"]["Code"] == "404":
```

Comment:
what's 404? can we avoid using plain text? we can use something like status code from django or requests. Something like it